share/store/eds_file.go

```diff
@@ -15,11 +14,23 @@ type EdsFile interface {
 	// Size returns square size of the file.
 	Size() int
 	// Share returns share and corresponding proof for the given axis and share index in this axis.
-	Share(ctx context.Context, axisType rsmt2d.Axis, axisIdx, shrIdx int) (share.Share, nmt.Proof, error)
+	Share(ctx context.Context, x, y int, proofType ProofAxis) (*share.ShareWithProof, error)
```

Comment:
As we discussed, the file/storage itself should be responsible for the decision of which proof to return, not the user.

Reply:
Imagine if only Col is available but we request the Row here.

Reply:
>Imagine if only Col is available but we request the Row here.
Exactly for case, when specific proof is required this option is created. To allow caller to request explicit axis proofs if needed (for example for befp building).`But user is not forced to, and if he wants to allow implementation to choose there is `proofAxisAny` option.

For sampling protocol you would need to use only `proofAxisAny` option.

Reply:
I think befps will have to be generated inside of the file and not outside. If not befp, there is no other usecase needing to request specific proof.

Reply:
befps are happening on sampling (protocol level), during eds reconstruction. So should be generated on protocol level too, using api provided by store. There is no need to put it inside File reading / writing logic

Reply:
I can remove no problem if you are sure if befp will be inside Store

Reply:
I got to think about this more and I don't think befps will anyhow be part of the file/store, nor anyone will need to ask the store for the proofs in case of befps.

Reply:
Befp will be formed in runtime when we fetch a square. That bad square might be stored afterwards but bepf will be already formed and stored separately. And when it's formed we would only use pieces coming from the network.

Reply:
> So generation logic should happen on protocol layer too, utilising api provided by Store

So I don't see how the protocol would need to access the file for the proof when protocol already has all the proofs in memory or fetches they from the network.

Reply:
Removed the option to select axis of proof.  Befp is a bit too far, so will change later if we need to.